experiments/test_helpers.rb

```diff
@@ -214,6 +228,12 @@ def wait_for_completion
         @request_threads.each(&:join)
         @monitor_thread.join if @is_adaptive
         @end_time = Time.now
+
+        # Clean up per-thread resources
```

Comment:
not saying it's bad, but I'm interested to know whether you saw something that made you want to add a cleanup